feat(aws): Adds support for Elasticache.

robh007 opened the pull request:
**Objective**:

Adds support for ```elasticahe_cluster``` & ```elasticache_replication_group```

** Closes ** #105 

**Example output:**

```
  NAME                                                    MONTHLY QTY  UNIT       PRICE   HOURLY COST  MONTHLY COST  

  aws_elasticache_cluster.this                                                                                       
  ├─ Elasticache (on-demand, cache.m4.large)                      730  hours      0.1560       0.1560      113.8800  
  └─ Elasticache snapshot storage                                   0  GB-months  0.0850       0.0000        0.0000  
  Total                                                                                        0.1560      113.8800  
                                                                                                                     
  aws_elasticache_replication_group.cluster                                                                          
  └─ Elasticache (on-demand, cache.m4.large)                   11,680  hours      0.1560       2.4960    1,822.0800  
  Total                                                                                        2.4960    1,822.0800  
                                                                                                                     
  aws_elasticache_replication_group.non-cluster                                                                      
  └─ Elasticache (on-demand, cache.m4.large)                    2,190  hours      0.1560       0.4680      341.6400  
  Total                                                                                        0.4680      341.6400  
                                                                                                                     
  aws_elasticache_replication_group.non-cluster-snapshot                                                             
  ├─ Elasticache (on-demand, cache.m4.large)                   11,680  hours      0.1560       2.4960    1,822.0800  
  └─ Elasticache snapshot storage                                   0  GB-months  0.0850       0.0000        0.0000  
  Total                                                                                        2.4960    1,822.0800  
                                                                                                                     
  OVERALL TOTAL (USD)                                                                          5.6160    4,099.6800  

```

**Pricing details:**

The elasticache service provides 2 pricing components described in https://aws.amazon.com/elasticache/pricing/. This adds support for the Elasticache clusters & replication_groups.

alikhajeh1 reviewed (commented):
Thanks @robh007! I left my suggestions, they also apply to `aws_elasticache_replication_group`.

Also, what happened to this discussion:
> Rob I've added support for elasticache cluster. However with elasticache cluster you can also create an elasticache_replcation_group. So in terraform there's two resources, the cluster can reference the replication group. So the cost is associated with the replication group. I was thinking that I add the elasticache_replication_group as another supported resource. If the cluster specifies the replication group the cost will be shown in the replication group & not the cluster. Does that make sense? I looked at what DynamoDB does with Global tables, but that's slightly different as the dynamoDB resource supports global tables & there's also an actual resource for global_tables.
> Ali: Yeah I think that makes sense, so if aws_elasticache_cluster.replication_group_id is set, it gets no costs and the aws_elasticache_replication_group.number_cache_clusters/node_type are used to calculate the cost, right?
> Rob: Yeah that appears to be correct. It's only applicable for Redis clusters.

robh007 commented:
@alikhajeh1 I've updated inline with your comments. Plus I've added the reference to the ```replication_group_id``` if present within the cluster configuration.

alikhajeh1 commented:
Closing this as https://github.com/infracost/infracost/pull/222 has an updated/refactored version
